### ISSUE TITLE
[FEATURE]: Add Advanced Options to Menu Bar

### ISSUE THREAD

**Issue by warpling** [raspberrypi/rpi-imager#955]
### Is your feature request related to a problem? Please describe.

When I first flashed my SD card from a custom image I was asked what advanced options I wanted to configure. When I later flashed a more standard image I wasn't so I thought it wasn't possible to configure things like SSID. I needed this for my headless setup but because there was no button or even menu bar item for Advanced Options I assumed they were unavailable.

### Describe the solution you would like to see implemented

There should be a button for accessing advanced options OR if there is a desire to keep most people away from them, only place it in the menu. A more advanced UX idea would be to permanently show the "Advanced Options" in the UI if the user has ever used them. 

### Describe alternatives you've considered

As smaller pis become popular so will headless configuration thus it should become easier not harder to set this up :) 

### Additional context

I was going to attach the app version of the mac app doesn't appear to have an About item in the menu bar either! These little things would go a long way to making the new user experience in 2024 better :)

### Version

1.9.2 (Default)

**Comment by lurch** [comment]
See also the related comments on #941 and #213

**Comment by warpling** [comment]
@lurch oh woah, does this mean customization is ignored for "non-custom" images?

**Comment by lurch** [comment]
Nah, it means that the OS's metadata in https://downloads.raspberrypi.com/os_list_imagingutility_v4.json explicitly needs to specify what kind of customisation the image wants.

**Comment by warpling** [comment]
and if the metadata doesn't then the settings config screen won't display but the set settings themselves are still applied?

**Comment by lurch** [comment]
If the metadata doesn't specify any customisation, then this dialogue won't appear:
![Screenshot from 2024-10-30 14-37-46](https://github.com/user-attachments/assets/06a48fb2-76a3-4238-a41c-eea8c72bdc1c)
and no settings will be applied to the image.

**Comment by warpling** [comment]
Ahhh okay, even if I bring up the customize screen with the keyboard shortcut and load a custom image?

**Comment by tdewey-rpi** [comment]
If you use the debug shortcut to open the OS customisation window against an OS that does not typically offer customisation, this is an unsupported action.

The customisation script will be written - but may have some, no, or a completely adverse effect.

I will be constraining the keyboard shortcut to remove that option on official builds, because it simply cannot offer a consistently good experience.

**Comment by warpling** [comment]
Hey @tdewey-rpi, that's good to know that but as someone brand new to pis and still learning how they work I assumed all OSes respected these conf settings at a very low level; it may be safe to assume others cobbling tutorials together, trying to get their pi to connect to wifi out of the box may then too.

I would guess that adding this option to the menu bar and greying it out when official builds are selected would communicate what's going on to users much more clearly than simply disabling the hotkey and making people wonder why sometimes the settings pop-up and sometimes they don't during certain installs. What do you think?

**Comment by tdewey-rpi** [comment]
> Hey @tdewey-rpi, that's good to know that but as someone brand new to pis and still learning how they work I assumed all OSes respected these conf settings at a very low level; it may be safe to assume others cobbling tutorials together, trying to get their pi to connect to wifi out of the box may then too.
> 
> I would guess that adding this option to the menu bar and greying it out when official builds are selected would communicate what's going on to users much more clearly than simply disabling the hotkey and making people wonder why sometimes the settings pop-up and sometimes they don't during certain installs. What do you think?

Thanks for the suggestion, @warpling

Your assumption is exactly what I want to avoid violating - and is a completely fair one to have made. I can only apologise that `rpi-imager` has fallen short in this respect.

There's another issue in this repository where I've outlined my plan for communicating what's going on - but essentially I want to add another button to the UI that will be grey/enabled as you select an OS that supports customisation, and include this in the work to make customisation more robust.

I wouldn't expect to see a release where you didn't get the entire feature - so the hotkey would not be disabled without having a replacement mechanism.

**Comment by warpling** [comment]
That sounds like a great plan! 

Thanks for making getting into this world easier and for the quick replies!

**Comment by tdewey-rpi** [comment]
This is effectively implemented in 2.0: customisation is shown only when appropriately tagged, and this makes configuring things like WiFi much more explicit as it's part of the main 'flow'.
